arch/arc/core/fault_s.S

```diff
@@ -39,7 +39,11 @@ GTEXT(z_irq_do_offload);
 #endif
 
 /* the necessary stack size for exception handling */
-#define EXCEPTION_STACK_SIZE 384
+#define EXCEPTION_STACK_SIZE 768
+
+#if EXCEPTION_STACK_SIZE > (CONFIG_ISR_STACK_SIZE >> 1)
```

Comment:
Out of curiosity: is that a common requirement to have IRQ stack at least twice larger than what exception may use? Then maybe we may implement that in generic code?

Reply:
no such a common requirement. It's ARC specific, the exception handling  uses the top of interrupt stack, if it uses too much stack, the interrupt handling will be impacted especially for nest interrupt handling. 

Why not allocate a separate stack for exception handling, because exception handling is not frequent,  if a separate stack is allocated, in normal situation, this stack won't be used. So for better mem footprint,  we use a fixed part of top of interrupt stack for exception handling.